includes/validation/class-amp-validation-manager.php

```diff
@@ -2178,7 +2178,7 @@ public static function enqueue_block_validation() {
 		$editor_support = Services::get( 'editor.editor_support' );
 
 		// Block validation script uses features only available beginning with WP 5.6.
-		if ( ! $editor_support->editor_supports_amp_block_editor_features() ) {
+		if ( ! $editor_support->is_current_screen_supported_block_editor_for_amp_enabled_post_type() ) {
 			return; // @codeCoverageIgnore
 		}
```

Comment:
Per above, this could then be:

```php
// Block validation script uses features only available beginning with WP 5.6.
$dependency_support = Services::get( 'dependency_support' );
if ( ! $dependency_support->has_support() ) {
	return; // @codeCoverageIgnore
}

// Only enqueue scripts on the block editor for AMP-enabled posts.
$editor_support = Services::get( 'editor.editor_support' );
if ( ! $editor_support->is_block_editor_screen_for_amp_enabled_post_type() ) {
	return;
}
```